### PR TITLE
doc: More detail on api documentation

### PR DESCRIPTION
**Pull request by mprevide**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)
dojot/dojot#1457

* **Other information**:

The history APi is not well designed, I tried to describe all its particularities for the user in the documentation, however such details do not work well with the automatic dreed tests. So, I decided to create 2 files, one to generate the documentation and one for the tests.
There are problems like https://github.com/apiaryio/api-blueprint/issues/120,  etc.

### CODE REVIEW

**Comment by mprevide** [comment]
I will create a PR for dev and 0.5 after this PR is merged.

File: docs/just-for-testing.apib [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+`?lastN=3&attr=temperature,pressure` will return the last 3 values from both
+`temperature` and `pressure` attributes.
+
+If no query string is set, then all values from all attributes are returned.
+
++ Parameters
+    + device_id:labtemp (required, string) - Device identifier
+    + attr:temperature (optional, string) - Device attribute to be
+    + dateFrom:`2018-06-05T18:00:00Z` (optional, string) - Start time of a time-based query
+    + dateTo:`2018-06-15T18:00:00Z` (optional, string) - End time of a time-based query
```
Comment (eduardogmisiuk) [inline docs/just-for-testing.apib:23]: In the `history.apib` file you changed the dates and the device ID. Wouldn't it be better to keep these files in sync?

File: docs/history.apib [hunk 2]
```diff
@@ -66,15 +73,165 @@ If no query string is set, then all values from all attributes are returned.
...
+            }
+
+## Filter by last N values and somes attributes  [/device/{device_id}/history?lastN={lastN}&attr={attr}&dateFrom={dateFrom}&dateTo={dateTo}]
+
++ Parameters
+    + device_id:b374a5 (required, string) - Device identifier
+    + attr:temperature,pressure (optional, string) - Device attribute to be requested. If not used, returns all attributes.
+    + dateFrom:`2018-06-05T18:00:00Z` (optional, string) - Start time of a time-based query
+    + dateTo:`2018-06-15T18:00:00Z` (optional, string) - End time of a time-based query
+    + lastN:3 (optional, number) - Number of most current values (valid for each attribute, if more than one)
```
[the remark this answers is not in the document]
Comment (mprevide) [reply docs/history.apib:105]: LastN? I'll try to make this clearer

File: docs/history.apib [hunk 3]
```diff
@@ -16,14 +18,19 @@ string might select more than one attribute. For instance:
...
+
+## Filter by last N values and only one attribute  [/device/{device_id}/history?lastN={lastN}&attr={attr}&dateFrom={dateFrom}&dateTo={dateTo}]
+
 + Parameters
-    + device_id:labtemp (required, string) - Device identifier
-    + attr:temperature (optional, string) - Device attribute to be
+    + device_id:b374a5 (required, string) - Device identifier
+    + attr:temperature (optional, string) - Device attribute to be requested. If not used, returns all attributes. (It can be a comma-separated list of attributes.)
     + dateFrom:`2018-06-05T18:00:00Z` (optional, string) - Start time of a time-based query
     + dateTo:`2018-06-15T18:00:00Z` (optional, string) - End time of a time-based query
```
Comment (eduardogmisiuk) [inline docs/history.apib:28]: The dates are different from the examples. The same happens in the other files.

**Pull request merged**